### DISCUSSION TITLE
Built project crashes the editor

### DISCUSSION THREAD

**Discussion by romanstetsenko** [nxrighthere/UnrealCLR#175] [Help] [answered]
Hi.
If `try...with` expression is used in any part of F# code, it crashes the editor.
Message from Crash Reporter:
```
UE4Editor_UnrealCLR!UnrealCLR::Module::OnWorldPostInitialization() [D:\games\Epic Games\MyProject\Plugins\UnrealCLR\Source\UnrealCLR\Private\UnrealCLR.cpp:1352]
UE4Editor_UnrealCLR!TBaseRawMethodDelegateInstance<0,UnrealCLR::Module,void __cdecl(UWorld *,UWorld::InitializationValues),FDefaultDelegateUserPolicy>::ExecuteIfSafe() [D:\games\Epic Games\UE_4.26\Engine\Source\Runtime\Core\Public\Delegates\DelegateInstancesImpl.h:469]
```

Code for reproducing:
```
namespace Game

open System.Drawing
open UnrealEngine.Framework

module Main = // Indicates the main entry point for automatic loading by the plugin
    let OnWorldBegin() = 
        try 
            Debug.AddOnScreenMessage(-1, 10.0f, Color.DeepPink, "Hello, Unreal Engine!")
        with e -> Debug.AddOnScreenMessage(-1, 10.0f, Color.Red, e.Message)
```
Also, I created a separate module with the exception handling expression and called it from the `module Main` - the editor crashed.
Then, I tried to call the separate module from an entry point in C# with the same result :(

**Comment by nxrighthere** [comment]
What platform do you use?

**Comment by romanstetsenko** [comment]
Edition	Windows 10 Pro N
Version	21H1
OS build	19043.1110
Experience	Windows Feature Experience Pack 120.2212.3530.0

**Comment by nxrighthere** [comment]
I'm unable to reproduce this.

How do you compile your project? Could you send me source files?

**Comment by romanstetsenko** [comment]
Nothing fancy, I use `dotnet build`.
Plugin is downloaded from here `https://github.com/nxrighthere/UnrealCLR/archive/refs/tags/1.29.zip`. Installation was completed succsesfully according to the [https://github.com/nxrighthere/UnrealCLR#auto](https://github.com/nxrighthere/UnrealCLR#auto)
All C# tests pass w/o issues.
Unreal project is very default `Games -> Blank -> C++ With Starter Content for Desktop`
Level: `/Game/StarterContent/Maps/Minimal_Default.Minimal_Default'`
Output Log after editor is loaded:
```
LogPluginManager: Mounting plugin UnrealCLR
LogUnrealCLR: Display: UnrealCLR::Module::StartupModule: Host path set to "d:/games/Epic Games/prj/MyProject/Plugins/UnrealCLR/Runtime/Win64/host/fxr/5.0.6/hostfxr.dll"
LogUnrealCLR: Display: UnrealCLR::Module::StartupModule: Host library loaded successfuly!
LogUnrealCLR: Display: UnrealCLR::Module::StartupModule: Host functions loaded successfuly!
LogUnrealCLR: Display: UnrealCLR::Module::StartupModule: Host runtime assembly loaded succesfuly!
LogUnrealCLR: Display: UnrealCLR::Module::StartupModule: Host runtime assembly initialized succesfuly!
LogUnrealCLR: Display: UnrealCLR::Module::StartupModule: Host loaded succesfuly!
```
[MyDotNetTest.zip](https://github.com/nxrighthere/UnrealCLR/files/6988519/MyDotNetTest.zip)

**Comment by nxrighthere** [comment, marked as answer]
> Nothing fancy, I use dotnet build.

You should [publish](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish) the project as stated in the [manual](https://github.com/nxrighthere/UnrealCLR/blob/master/MANUAL.md#entry-point) instead of just building it.

**Comment by romanstetsenko** [comment]
Oh, my. It was stupid of me. Thank you for the help.

**Reply by nxrighthere** [reply]
I'll try to handle such cases on the plugin side to avoid confusion.

**Reply by nxrighthere** [reply]
Resolved in https://github.com/nxrighthere/UnrealCLR/commit/60315d69d85c454b5537384319fab76a24bd5dde.
